x/escrow/init.go

```diff
@@ -0,0 +1,62 @@
+package escrow
+
+import (
+	"encoding/hex"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/x/cash"
+	"github.com/pkg/errors"
+)
+
+var _ weave.Initializer = (*Initializer)(nil)
+var burnAddress, _ = hex.DecodeString("0000000000000000000000000000000000000000")
+
+// Initializer fulfils the Initializer interface to load data from the genesis file
+type Initializer struct {
+	Minter cash.CoinMinter
+}
+
+// FromGenesis will parse initial escrow  info from genesis and save it in the database.
+func (i *Initializer) FromGenesis(opts weave.Options, db weave.KVStore) error {
+	var escrows []struct {
+		Sender    weave.Address   `json:"sender"`
+		Arbiter   weave.Condition `json:"arbiter"`
```

Comment:
Not sure about de-marshalling arbiter, I think Condition doesn't lend itself like Address...

There are a few lines in `Address.UnmarshalJSON()` that could me moved into `Condition.UnmarshalJSON()` to make this whole thing prettier...

```go
	case "cond":
		args := strings.Split(enc, "/")
		if len(args) != 3 {
			return errors.ErrInvalidInput.Newf("invalid condition format")
		}
		data, err := hex.DecodeString(args[2])
		if err != nil {
			return errors.ErrInvalidInput.Newf("malformed condition data: %s", err)
		}
		*a = NewCondition(args[0], args[1], data).Address()
		return nil
```

And have MarshalJSON() return the Stringified version of itself, so it is cleanly (de-)serializable

Reply:
Do I understand it correct that `MarshalJSON` should return `cond:foo/bar/HEXENCODED` and not the hex encoded string?

Reply:
Rather `foo/bar/HEXENCODED`